src/bepasty/tests/test_rest_server.py

```diff
@@ -0,0 +1,816 @@
+    check_err_response(response, 400)
+
+    # Content-Range: invalid (invalid format)
+    range_str = 'invalid'
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: bytes invalid (invalid format)
+    range_str = 'bytes invalid'
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: other 0-<sep - 1>/<len(data)> (invalid unit)
+    range_str = 'other {}-{}/{}'.format(0, sep - 1, len(UPLOAD_DATA))
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: bytes invalid-<sep - 1>/<len(data)> (invalid first)
+    range_str = 'bytes invalid-{}/{}'.format(sep - 1, len(UPLOAD_DATA))
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: bytes 0-invalid/<len(data)> (invalid last)
+    range_str = 'bytes {}-invalid/{}'.format(0, len(UPLOAD_DATA))
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: bytes <sep - 1>-0/<len(data)> (invalid first > last)
+    range_str = 'bytes {}-{}/{}'.format(sep - 1, 0, len(UPLOAD_DATA))
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: bytes 0-<sep - 1>/* (not supported)
+    range_str = 'bytes {}-{}/*'.format(0, sep - 1)
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: bytes */<len(data)> (not supported)
+    range_str = 'bytes */{}'.format(len(UPLOAD_DATA))
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # Content-Range: bytes */* (not supported)
+    range_str = 'bytes */*'
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_err_response(response, 400)
+
+    # upload first part of data
+    range_str = 'bytes {}-{}/{}'.format(0, sep - 1, len(UPLOAD_DATA))
+    response = _upload(client, data, token='full', filename=filename,
+                       ftype=ftype, range_str=range_str)
+    check_upload_response(response, 200)
+
+    # upload reminder of data
+    data = UPLOAD_DATA[sep:]
+    range_str = 'bytes {}-{}/{}'.format(sep, len(UPLOAD_DATA) - 1,
```

Comment:
shouldn't this be: `sep, sep + len(UPLOAD_DATA) - 1, len(UPLOAD_DATA)`

and if so, why did the test succeed? bug?

Reply:
ah, no, my fault. it would be sep + len(data) - 1, but your version is also correct.